examples/geometry_optimization.jl

```diff
@@ -0,0 +1,84 @@
+# # Geometry optimization
+#
+# We use the DFTK and Optim packages in this example to compute numerically the bond length
+# of the ``H_2`` molecule that minimizes the total energy. We setup ``H_2`` in an
+# LDA model just like in the [Tutorial](@ref) for silicon.
+using DFTK
+using Optim
+using LinearAlgebra
+using Printf
+
+kgrid = [1, 1, 1]       # k-Point grid
+Ecut = 5                # kinetic energy cutoff in Hartree
+tol = 1e-8              # tolerance for the optimization routine
+a = 10                  # lattice constant in Bohr
+lattice = a * Diagonal(ones(3))
+invlattice = inv(lattice)        # matrix to go from cartesian to reduced coordinates
+H = ElementPsp(:H, psp=load_psp("hgh/lda/h-q1"));
+
+# We define a blochwave and a density to be used as global variables where we
```

Comment:
I'd emphasise the "why": "... as global variables, such that we can transfer the solution from one iteration to another".